packages/react/src/alert-dialog/root/AlertDialogRoot.tsx

```diff
@@ -15,19 +16,22 @@ import { PortalContext } from '../../portal/PortalContext';
 const AlertDialogRoot: React.FC<AlertDialogRoot.Props> = function AlertDialogRoot(props) {
   const { children, defaultOpen = false, onOpenChange, open } = props;
 
-  const parentDialogRootContext = React.useContext(AlertDialogRootContext);
+  const parentAlertDialogRootContext = React.useContext(AlertDialogRootContext);
+  const parentDialogRootContext = React.useContext(DialogRootContext);
+
+  const parentContext = parentAlertDialogRootContext ?? parentDialogRootContext ?? undefined;
 
   const dialogRoot = useDialogRoot({
     open,
     defaultOpen,
     onOpenChange,
     modal: true,
     dismissible: false,
-    onNestedDialogClose: parentDialogRootContext?.onNestedDialogClose,
-    onNestedDialogOpen: parentDialogRootContext?.onNestedDialogOpen,
+    onNestedDialogClose: parentContext?.onNestedDialogClose,
```

Comment:
Wouldn't it be cleaner to have a shared context for the element that are shared across the various dialog types?
Or are we confident that we won't have another type of dialog (not sure if a drawer falls in the category) that we will have to remember to add here?

Reply:
Yes, I started working on this coincidently, let me open a PR. @mj12albert please assign an issue to you if you plan to work on it to avoid duplicated work.

Reply:
Alternative PR: https://github.com/mui/base-ui/pull/1167